sdk/cardboard.cc

```diff
@@ -343,6 +343,14 @@ void CardboardQrCode_destroy(const uint8_t* encoded_device_params) {
   }
   delete[] encoded_device_params;
 }
+void CardboardQrCode_setAndSaveDeviceParams(const uint8_t* uri, int size) {
+  if (CARDBOARD_IS_NOT_INITIALIZED() || CARDBOARD_IS_ARG_NULL(uri)
+      || size <= 0) {
```

Comment:
Since the CARDBOARD_IS_NOT_INITIALIZED and CARDBOARD_IS_ARG_NULL macros log when the assertion is true, we're not logging anything in the condition body.

Consider moving the `size` check to another conditional structure and logging when true.